source/02_algo/distributed_rl_eng.rst

```diff
@@ -0,0 +1,122 @@
+
+
+Research Direction
+---------
+
+System
+~~~~~~
+
+Overall Architecture
+^^^^^^^^^^
+For common decision problems, the two most commonly used distributed architectures are IMPALA [3]_ and SEED RL [4]_.
+
+.. image:: ./images/impala.png
+  :align: center
+  
+- The former is the classic Actor-Learner mode; that is, the data collection and training sides are entirely separated, and the latest neural network model is regularly passed from the Learner to the Actor, and the observations collected by the Actor are sent to the Learner after collecting a certain amount of data (i.e. observations). If there are multiple Learners, they also periodically synchronize their gradients to update the neural network (i.e. the data-parallel model in distributed deep learning).
+
+.. image:: ./images/seed_rl.png
+  :scale: 50%
+  :align: center
+
+- On the basis of the former, the latter is dedicated to optimizing the loss of the transmission model. SEED RL strips out the part used for inference to generate actions, and puts it together with the training end to update the model through efficient inter-TPU communication technology. The cost of passing the model in IMPALA is greatly reduced, and for cross-machine communication between the environment and the reasoning Actor, SEED RL uses an optimized gRPC scheme to pass the observation and action, so there is not much burden.
+
+.. note::
+There is no absolute superiority or inferiority between these two schemes. If it is more expensive to transmit models across machines and there are better communication components between GPU/TPU, then SEED RL could be a more suitable option. Otherwise, if it is more expensive to transmit observations and actions data across machines, then IMPALA is a more stable choice. In addition, IMPALA can accumulate a batch of data for data transmission, while SEED RL requires data transmission in each interactive frame. This is a classic data batch and stream processing comparison problem. For the current machine learning community, the former is generally more user-friendly. Also, if the entire training procedure requires a higher degree of freedom and customization, such as dynamically controlling some behavior of the Actor, IMPALA is more convenient.
+
+In addition to the above two architectures, there are many other distributed reinforcement learning design schemes, such as A3C [5]_ and Gossip A2C [6]_ that introduce asynchronous neural network update schemes. In order to support large-scale self-play, AlphaStar [1]_ with a complex League mechanism was designed, and MuZero [7]_ combined with model-based RL and MCTS-related modules will not be described here. Interested readers can refer to the specific Papers or refer to our `Algorithm Raiders Collection section <../12_policies/index_zh.html>`_.
+
+Single Point Efficiency Optimization
+^^^^^^^^^^^^^
```

Comment:
is there any more proper expression for `Single Point`?

Reply:
maybe “Single Efficiency Optimization” , “Point Focus Efficiency Optimization” or "Single Objective Efficiency Optimization”？ Choose one~

Reply:
done~